blackcellmagic.py

```diff
@@ -37,8 +37,12 @@ def black(self, line, cell):
         args = magic_arguments.parse_argstring(self.black, line)
         line_length = args.line_length
         if cell:
+            blackened = format_str(src_contents=cell, line_length=line_length)
```

Comment:
You are doing the work twice in line 40 and in line 45. And condition chaining in line 41,42 need not be nested. You can just do this and achieve the same.

```python
formatted = format_str(src_contents=cell, line_length=line_length)
if formatted and (formatted[-1] == "\n"):
    formatted = formatted[:-1]
self.shell.set_next_input(formatted, replace=True)
```